File: Backend/src/main/java/com/backend/model/Titulo.java
package com.backend.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.time.LocalDate;
import java.util.List;

@Data
@Entity
@Table(name = "titulo")
public class Titulo {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(length = 200, nullable = false)
    private String nome;

    @Column
    private LocalDate ano;

    @Column(length = 200, nullable = false)
    private String sinopse;

    @NotNull
    @Column
    private String categoria;

    @ManyToOne
    @JoinColumn(name = "idDiretor")
    private Diretor diretor;

    @ManyToOne
    @JoinColumn(name = "idClasse")
    private Classe classe;

    @ManyToMany
    @NotNull
    @JoinTable(name = "ator_titulo", 
            joinColumns = { @JoinColumn(name = "idTitulo") }, 
            inverseJoinColumns = { @JoinColumn(name = "idAtor") })
    private List<Ator> atores;
}

File: Frontend/src/app/titulo/model/titulo.ts
import { Ator } from "../../ator/model/ator";
import { Classe } from "../../classe/model/classe";
import { Diretor } from "../../diretor/model/diretor";

export interface Titulo {
    id: number;
    nome: string;
    ano: number;
    sinopse: string;
    categoria: string;
    diretor: Diretor;
    classe: Classe;
    atores: Ator[];
}
